src/libcxx.zig

```diff
@@ -238,21 +246,36 @@ pub fn buildLibCXXABI(comp: *Compilation) !void {
 
     const cxxabi_include_path = try comp.zig_lib_directory.join(arena, &[_][]const u8{ "libcxxabi", "include" });
     const cxx_include_path = try comp.zig_lib_directory.join(arena, &[_][]const u8{ "libcxx", "include" });
+    var c_source_files = std.ArrayList(Compilation.CSourceFile).init(arena);
+    try c_source_files.ensureCapacity(libcxxabi_files.len);
 
-    var c_source_files: [libcxxabi_files.len]Compilation.CSourceFile = undefined;
-    for (libcxxabi_files) |cxxabi_src, i| {
+    for (libcxxabi_files) |cxxabi_src| {
         var cflags = std.ArrayList([]const u8).init(arena);
 
-        try cflags.append("-DHAVE___CXA_THREAD_ATEXIT_IMPL");
+        if (target.os.tag == .wasi) {
+            if (std.mem.startsWith(u8, cxxabi_src, "src/cxa_thread_atexit.cpp"))
+                continue;
+
+            if (std.mem.startsWith(u8, cxxabi_src, "src/cxa_exception.cpp"))
+                continue;
+
+            if (std.mem.startsWith(u8, cxxabi_src, "src/cxa_personality.cpp"))
+                continue;
+
+            try cflags.append("-fno-exceptions");
+        } else {
+            try cflags.append("-DHAVE___CXA_THREAD_ATEXIT_IMPL");
+            try cflags.append("-D_LIBCPP_ENABLE_CXX17_REMOVED_UNEXPECTED_FUNCTIONS");
+        }
+
         try cflags.append("-D_LIBCPP_DISABLE_EXTERN_TEMPLATE");
-        try cflags.append("-D_LIBCPP_ENABLE_CXX17_REMOVED_UNEXPECTED_FUNCTIONS");
         try cflags.append("-D_LIBCXXABI_BUILDING_LIBRARY");
         try cflags.append("-D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS");
         try cflags.append("-D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS");
         try cflags.append("-fvisibility=hidden");
         try cflags.append("-fvisibility-inlines-hidden");
 
-        if (target.abi.isMusl()) {
+        if (target.abi.isMusl() or target.os.tag == .wasi) {
```

Comment:
I don't think an explicit check for WASI here is necessary since WASI's ABI is musl: [lib/std/target.zig#L465](https://github.com/ziglang/zig/blob/d2427e6490587c4d9cfb1b2dfb735c20967121db/lib/std/target.zig#L465)